core/src/common.cpp

```diff
@@ -397,23 +827,43 @@ namespace irods::http
 			// It's possible that the admin didn't include the OIDC configuration stanza.
 			// This use-case is allowed, therefore we check for the OIDC configuration before
 			// attempting to access it. Without this logic, the server would crash.
-			const auto oidc_config_iter =
-				config.find(nlohmann::json::json_pointer{"/http_server/authentication/openid_connect"});
-			if (oidc_config_iter == std::end(config)) {
+			if (static const auto oidc_conf_exists{config.contains(nlohmann::json::json_pointer{"/http_server/authentication/openid_connect"})}; !oidc_conf_exists) {
 				logging::debug("{}: No 'openid_connect' stanza found in server configuration.", __func__);
 				logging::error("{}: Could not find bearer token matching [{}].", __func__, bearer_token);
 				return {.response = fail(status_type::unauthorized)};
 			}
 
 			// If we're running as a protected resource, assume we have a OIDC token
-			if (oidc_config_iter->at("mode").get_ref<const std::string&>() == "protected_resource") {
-				body_arguments args{{"token", bearer_token}, {"token_type_hint", "access_token"}};
+			if (irods::http::globals::oidc_configuration().at("mode").get_ref<const std::string&>() ==
+			    "protected_resource") {
+				nlohmann::json json_res;
+
+				// Try parsing token as JWT Access Token
+				try {
+					auto token{jwt::decode<jwt::traits::nlohmann_json>(bearer_token)};
+					auto possible_json_res{validate_using_local_validation(token)};
+
+					if (possible_json_res) {
+						json_res = *possible_json_res;
+					}
+				}
+				// Parsing of the token failed, this is not a JWT access token
+				catch (const std::exception& e) {
+					logging::info("{}: {}", __func__, e.what());
+				}
```

Comment:
Would lowering it further to trace be fine? Since this is an artifact of design, I feel like it's likely to be a low priority log.

Reply:
trace is usually only for letting a reader know the flow... no values shown/logged.

`debug` here feels... plenty?

Reply:
Either way. `info` just seemed a bit much if this is a common occurrence. On the other hand, maybe it's just right. I'm not familiar enough with the feature to know better.

Reply:
Agreed.   Not sure how often someone starts a flow with a bad token.

Being HTTP, I wouldn't be surprised if scripts are always trying shenanigans to get in... so that would suggest `debug`...   but if that's not the case, yes, `info` feels correct enough.

Reply:
Logging as debug.